ruleset.go

```diff
@@ -127,15 +98,76 @@ type Signature struct {
 	ParamTypes map[string]string
 }
 
-// NewSignature returns the Signature of the given ruleset.
-func NewSignature(rs *Ruleset) *Signature {
-	pt := make(map[string]string)
-	for _, p := range rs.Params() {
-		pt[p.Name] = p.Type
+// NewSignature create
+func NewSignature() *Signature {
+	return &Signature{
+		ParamTypes: make(map[string]string),
 	}
+}
 
-	return &Signature{
-		ParamTypes: pt,
-		ReturnType: rs.Type,
+// StringP adds a string param to the signature.
+func (s *Signature) StringP(name string) *Signature {
```

Comment:
My inner lisper automatically assumed "StringP" would return a boolean :-)   P is a common short-hand for "predicate", and in Lisp that generally means type predicate and thus `(string-p "hello world") -> true`

Reply:
For clarity - I'm not asking you to do anything here, just recording my surprise ;-)